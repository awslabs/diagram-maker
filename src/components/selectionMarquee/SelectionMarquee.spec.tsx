import * as Preact from 'preact';
import { shallow } from 'preact-render-spy';

import { SelectionMarquee } from '.';

describe('SelectionMarquee', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders at the given position and size', () => {
    const selectionMarquee = shallow(
      <SelectionMarquee
        anchor={{ x: 200, y: 100 }}
        position={{ x: 800, y: 500 }}
      />
    );

    expect(selectionMarquee).toMatchSnapshot();
  });
});
