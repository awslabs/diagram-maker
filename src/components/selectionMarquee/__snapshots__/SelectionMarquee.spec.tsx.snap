// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionMarquee renders at the given position and size 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="dm-marquee"
  style="height: 400px; left: 200px; top: 100px; width: 600px;"
  data-event-target={true}
  data-type="DiagramMaker.SelectionMarquee"
>
</div>

`;
